projects/plugins/jetpack/tests/php/_inc/lib/test_class.rest-api-endpoints.php

```diff
@@ -1126,4 +1126,103 @@ public function test_get_user_connection_data_without_master_user() {
 		$response_data = $response->get_data();
 		$this->assertNull( $response_data['connectionOwner'] );
 	}
+
+	/**
+	 * Test fetching a site's purchase token.
+	 *
+	 * @since 9.8.0
+	 */
+	public function test_get_purchase_token() {
+		$purchase_token = '1ApurchaseToken1';
+		Jetpack_Options::update_option( 'id', 1234 );
+		Jetpack_Options::update_option( 'purchase_token', $purchase_token );
+
+		// Create a user and set it up as current.
+		$user = $this->create_and_get_user( 'administrator' );
+		wp_set_current_user( $user->ID );
+
+		// Fetch purchase token.
+		$response = $this->create_and_get_request( 'purchase-token', array(), 'GET' );
+
+		// Confirm purchase token exists.
+		$this->assertResponseStatus( 200, $response );
+		$this->assertEquals( $purchase_token, $response->get_data() );
+	}
+
+	/**
+	 * Test fetching a site's purchase token when no site is registered.
+	 *
+	 * @since 9.8.0
+	 */
+	public function test_get_purchase_token_no_site_registered() {
+		$purchase_token = '1ApurchaseToken1';
+		Jetpack_Options::update_option( 'purchase_token', $purchase_token );
+
+		// Create a user and set it up as current.
+		$user = $this->create_and_get_user( 'administrator' );
+		wp_set_current_user( $user->ID );
+
+		// Fetch purchase token.
+		$response = $this->create_and_get_request( 'purchase-token', array(), 'GET' );
+
+		// Confirm that the request failed.
+		$this->assertResponseStatus( 500, $response );
+		$this->assertResponseData( array( 'code' => 'site_not_registered' ), $response );
+	}
+
+	/**
+	 * Test deleting a site's purchase token.
+	 *
+	 * @since 9.8.0
+	 */
+	public function test_delete_purchase_token() {
```

Comment:
Could we add two more cases here pls?
- Same endpoint, blog token -> should succeed
- Same endpoint, user without permissions -> should fail